src/server/rdb_load.cc

```diff
@@ -893,14 +954,73 @@ void RdbLoaderBase::OpaqueObjLoader::HandleBlob(string_view blob) {
       res->encoding = OBJ_ENCODING_INTSET;
     }
     pv_->ImportRObj(res);
-  } else if (rdb_type_ == RDB_TYPE_HASH_ZIPLIST) {
-    unsigned char* lp = lpNew(blob.size());
-    if (!ziplistPairsConvertAndValidateIntegrity((const uint8_t*)blob.data(), blob.size(), &lp)) {
-      LOG(ERROR) << "Zset ziplist integrity check failed.";
-      zfree(lp);
+  } else if (rdb_type_ == RDB_TYPE_SET_LISTPACK) {
+    if (!lpValidateIntegrityAndDups((uint8_t*)blob.data(), blob.size(), 0, 1)) {
+      LOG(ERROR) << "ListPack integrity check failed.";
       ec_ = RdbError(errc::rdb_file_corrupted);
       return;
     }
+    unsigned char* lp = (unsigned char*)blob.data();
+    auto iterate_and_apply_f = [lp](auto f) {
+      for (unsigned char* cur = lpFirst(lp); cur != nullptr; cur = lpNext(lp, cur)) {
+        unsigned int* slen = nullptr;
+        long long* lval = nullptr;
+        unsigned char* res = lpGetValue(cur, slen, lval);
+        f(res, slen, lval);
+      }
+    };
+    const bool use_set2 = GetFlag(FLAGS_use_set2);
+    robj* res = nullptr;
+    if (use_set2) {
+      StringSet* set = new StringSet{CompactObj::memory_resource()};
+      res = createObject(OBJ_SET, set);
+      res->encoding = OBJ_ENCODING_HT;
+      auto f = [this, res](unsigned char* val, unsigned int* slen, const long long* lval) {
+        sds sdsele = (val) ? sdsnewlen(val, *lval) : sdsfromlonglong(*lval);
+        if (!((StringSet*)res->ptr)->AddSds(sdsele)) {
+          LOG(ERROR) << "Error adding to member set22";
+          ec_ = RdbError(errc::duplicate_key);
+        }
+      };
+      iterate_and_apply_f(f);
+    } else {
+      res = createSetObject();
+      auto f = [this, res](unsigned char* val, unsigned int* slen, const long long* lval) {
+        sds sdsele = (val) ? sdsnewlen(val, *lval) : sdsfromlonglong(*lval);
+        if (!dictAdd((dict*)res->ptr, sdsele, nullptr)) {
```

Comment:
This is a little bit weird for me, it's used in the `CreateSet` and should be correct, but I am not 100% sure why the  last arguments  is nullptr. (Maybe that means that the hashtable is used as a set, which should be probably the case)